Library/Homebrew/formula.rb

```diff
@@ -920,6 +920,14 @@ def bash_completion
     prefix/"etc/bash_completion.d"
   end
 
+  # The directory where the formula's Bash completion files should be
+  # installed that depend on `bash-completion@2` or Bash 4+.
+  # This is symlinked into `HOMEBREW_PREFIX` after installation or with
+  # `brew link` for formulae that are not keg-only.
+  def bash_completion2
+    share/"bash-completion/completions"
```

Comment:
Not sure on name since it is possible `bash-completion@2` will become `bash-completion@3`, ... in future.